### PR TITLE
test(Navigation/sidebar): visual regression test with cy

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
visual regression test with cy

### CODE REVIEW

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  });
+
+  it('displays the direct child items and highlights the item when hovered ', () => {
+    cy.get('li').eq(2).trigger('mouseover');
+    cy.get('[data-cy=collapse]').should('be.visible');
+
+    cy.matchImageSnapshot();
+  });
+
+  it('hilights the menu item when double clicked', () => {
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:60]:
```suggestion
  it('highlights the menu item when it is double clicked', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  it('hilights the parent item when the item is clicked', () => {
+    cy.get("[alt='mini logo']").click();
+    cy.get('li').eq(2).trigger('mouseover');
+    cy.get('[data-testid=ExpandMoreIcon]').should('be.visible');
+    cy.get('[data-cy=collapse]').as('collapse').should('be.visible');
+    cy.get('@collapse')
+      .find('ul')
+      .first()
+      .as('first_ele_collapse')
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:74]:
```suggestion
      .as('first_element_collapse')
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      .trigger('mouseover');
+    cy.get('@first_ele_collapse')
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:76]:
```suggestion
    cy.get('@first_element_collapse')
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      .find('[data-testid=ExpandMoreIcon]')
+      .should('be.visible');
+    cy.get('@first_ele_collapse')
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:79]:
```suggestion
    cy.get('@first_element_collapse')
```

File: cypress/support/commands.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { addMatchImageSnapshotCommand } from 'cypress-image-snapshot/command';
+
+addMatchImageSnapshotCommand({
+  capture: 'viewport',
+  customDiffConfig: { threshold: 0.1 },
+  customSnapshotsDir: './cypress/visual-testing-snapshots',
+  failureThreshold: 0.03,
```
[the remark this answers is not in the document]
Comment (Noha-ElAbrouki) [reply cypress/support/commands.ts:7]: this is the default configuration in their github repository https://github.com/jaredpalmer/cypress-image-snapshot

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      } else if (index === 1) {
+        cy.get('@element').contains('Home');
+      } else {
+        cy.get('@element').contains('Configuration');
+      }
+    });
+    cy.matchImageSnapshot();
+  });
+
+  it('displays the direct child items and highlights the item when hovered ', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:53]:
```suggestion
  it('displays the direct child items and highlights the item when hovered', () => {
```

**Review by bdauria — DISMISSED** [review]

**Review by Thebarda — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**